### PR TITLE
Hub resolver: add version constraints

### PR DESCRIPTION
**Pull request by chmouel**
This let the user specify a version constraint to choose from.

Version constraint looks like this:

```yaml
version: ">= 0.5.0"
```

This will only choose the tasks that are greater than 0.5.0

Additional constraint operators are available, for example:

```yaml
version: ">= 0.5.0, < 2.0.0"
```

/kind feature

This will only choose the tasks that are greater than 0.5.0 and less than 2.0.0

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [X] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [X] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [X] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [X] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [X] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [X] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
hub resolver can now specify a version constraint to choose a version (example: version: ">=0.2.0,< 1.0.0")
```

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: docs/hub-resolver.md [hunk 1]
```diff
@@ -126,6 +126,36 @@ spec:
...
+
+```yaml
+params:
+  - name: name
+    value: git-clone
+  - name: version
+    value: ">=0.7.0, < 2.0.0"
+```
+
+Will select the git-clone task that is greater than version `0.7.0` and less than version `2.0.0`
```
Comment (jerop) [inline docs/hub-resolver.md:153]: currently, we have up to 0.9.0 -- so three versions meet this constraint -- which version will be chosen? read through https://github.com/hashicorp/go-version but it wasn't clear to me

Comment (chmouel) [reply docs/hub-resolver.md:153]: ah yeah i meant to add that to the documentation but forgot, the code has some logic to only choose the latest one as tested here https://github.com/tektoncd/pipeline/pull/7257/files#diff-8867ae56005554e8425016bb15b437a7b2b17b9d763572ce20f0dc9c974ec297R247

will amend the doc,

**Comment by chmouel** [comment]
I used [go-version](https://github.com/hashicorp/go-version) since the dependency was already there (as indirect dep) there is as well [semver](https://github.com/Masterminds/semver) which seems as good (but better documented) but not having to add a new deps to maintain/tracks wins everything imho :)

**Review by jerop — APPROVED** [review]
/lgtm

🎉

**Pull request merged**